### ISSUE TITLE
High CPU usage

### ISSUE THREAD

**Issue by linuz90** [getcursor/cursor#918]
1) If you can, please include a screenshot of your problem
2) Please include the name of your operating system
3) If you can, steps to reproduce are super helpful

I'm using Cursor on an M1 MacBook Pro (2020). Imported settings from VSC.

After less than 1h of usage, Cursor starts using the CPU like crazy, to the point I start hearing my MPB fan loudly spinning (for the first time since I bought it).

<img width="422" alt="CleanShot 2023-09-12 at 09 26 53@2x" src="https://github.com/getcursor/cursor/assets/9629188/66bf1c6a-a6f0-4b9b-8ed2-0c14d7509c5c">

I tried disabling as many extensions as possible, I've made sure Cursor is updated, but nothing works.

Both VSC and VSC Insiders work flawlessly.

**Comment by truell20** [comment]
I assume this never happens for you on VSCode?

**Comment by linuz90** [comment]
> I assume this never happens for you on VSCode?

Correct, never. I also noticed a teammate yesterday had the same issue with Cursor, but on the other hand it's been reducing in the last few days on my Mac, not sure why

**Comment by BenCully** [comment]
Update: this issue seems to be fixed in the newest release of cursor. Also for more context this was happening while using local and liveshare modes.

**Comment by afarwind** [comment]
I'm using Cursor on an M3 MacBook Pro (2023). Imported settings from VSC.
I am also having this issue

**Comment by ufukomer** [comment]
same here. Macbook m1 processor:

<img width="1072" alt="image" src="https://github.com/user-attachments/assets/9eedf8fd-35de-460a-bf65-2e5d78811076">

**Comment by sachajw** [comment]
This is happening for me too, which basically makes the Ai functionality impossible to use. I am using a MacBook Pro M2 with Sonoma 14.6.1

**Comment by aalyokhin** [comment]
The same issue for me, M1 Pro runs hot with audible fan noise.

![image](https://github.com/user-attachments/assets/2da526d3-2826-4503-8398-ab988dcc5517)
<img width="400" alt="SCR-20240916-lkus" src="https://github.com/user-attachments/assets/670a805d-3933-4b27-902a-82ac660c2601">


I read somewhere that it may be caused by the ESLint extension. So I've disabled it and got this:
<img width="400" alt="SCR-20240916-lmze" src="https://github.com/user-attachments/assets/88c5c001-ea0b-4617-955c-9ca21ca421c3">
<img width="400" alt="SCR-20240916-lmfy" src="https://github.com/user-attachments/assets/6179ef0b-6250-4595-90d2-fd52016d0453">

So it helped but actually didn't.

**Comment by m-w-d-d** [comment]
Can confirm getting the same issue as the above - iMac 2019 3.6 GHz 8-Core Intel Core i9 - issue never happens with VS Code

<img width="778" alt="image" src="https://github.com/user-attachments/assets/1392b625-65e0-4072-bc49-44739ea3de3c">

**Comment by ufukomer** [comment]
@m-w-d-d my suggestion; terminate this process and meanwhile check the error message on your ide below that it will state which plugin is causing this, then remove it. I still couldn't figure it out but ide is working like a charm even when this process is terminated.

**Comment by m-w-d-d** [comment]
> @m-w-d-d my suggestion; terminate this process and meanwhile check the error message on your ide below that it will state which plugin is causing this, then remove it. I still couldn't figure it out but ide is working like a charm even when this process is terminated.

Will have a try with this @ufukomer - thanks for the suggestion!

**Comment by ufukomer** [comment]
Here is more clue for you. When I terminate the process with high cpu usage, this text on the bottom of the IDE is appearing: **"Initializing JS/TS language features The extension host terminated unexpectedly. Restarting...**" This is not about the extensions I have installed. Because I removed all of them and tried it. This is about in-built extensions I guess.

<img width="1440" alt="Screenshot 2024-10-24 at 16 06 51" src="https://github.com/user-attachments/assets/0be262ca-274d-4f67-beee-b52b26f22ece">

@truell20

**Comment by rstupa** [comment]
Same problem already 2 month, so I using another ide. MacBook Air m2 8gb 15.2 (24C101)

[Sample of Cursor Helper (Plugin): extension-host \[2-4\].txt](https://github.com/user-attachments/files/18547971/Sample.of.Cursor.Helper.Plugin.extension-host.2-4.txt)

<img width="1181" alt="Image" src="https://github.com/user-attachments/assets/463d7dff-dde6-45fa-801a-34473d93390e" />

**Comment by gardner** [comment]
This happens often. Generally it goes like this:

* I notice the laptop is nuclear hot
* I look in activity monitor and see Cursor chewing through resources
* I notice my battery is almost dead
* I quick Cursor

**Comment by rstupa** [comment]
[**[](url)**](https://forum.cursor.com/t/cursor-helper-plugin-is-killing-my-m1-cpu/2305/16)


Read this one

<img width="625" alt="Image" src="https://github.com/user-attachments/assets/ea0f8614-39bc-4f40-a6b8-7fdbd3489662" />

**Comment by bower-joby** [comment]
👆 This turned out to be my problem as well. Once I disabled this extension and restarted the extensions, the CPU-pegged process disappeared. For me, it was sticking around 200% on an MBP M4 running 15.3.x.

![Image](https://github.com/user-attachments/assets/8bef53ef-215b-4b54-b2e9-4b2d86d01707)

**Comment by webbower** [comment]
👆 This turned out to be my problem as well. Once I disabled this extension and restarted the extensions, the CPU-pegged process disappeared. For me, it was sticking around 200% on an MBP M4 running 15.3.x.

![Image](https://github.com/user-attachments/assets/8bef53ef-215b-4b54-b2e9-4b2d86d01707)

**Comment by tallytarik** [comment]
I was experiencing the same until I disabled this old [settings sync extension](https://marketplace.visualstudio.com/items?itemName=Shan.code-settings-sync). Any chance you're using that? 

Otherwise suggest trying [extension bisect](https://code.visualstudio.com/blogs/2021/02/16/extension-bisect) to see if it is caused by a different extension.
